osf/utils/datetime_aware_jsonfield.py

```diff
@@ -98,14 +101,19 @@ def get_prep_lookup(self, lookup_type, value):
 class DateTimeAwareJSONFormField(JSONFormField):
     def to_python(self, value):
         try:
-            return decode_datetime_objects(json.loads(value))
+            return decode_datetime_objects(value)
         except TypeError:
             raise ValidationError(
                 self.error_messages['invalid'],
                 code='invalid',
                 params={'value': value},
             )
 
+    def from_db_value(self, value, expression, connection):
+        if value is None:
+            return None
+        return super(DateTimeAwareJSONFormField).to_python(decode_datetime_objects(value))
+
```

Comment:
I learned that `JSONFormField` (i.e. `django.forms`) is very different from `JSONField` (i.e. `django.db.models`). Thus my understanding is that `DateTimeAwareJSONFormField` doesn't need this  `from_db_value()` at all. It is the `to_python` we want to update.

Before (1.11): https://docs.djangoproject.com/en/1.11/_modules/django/contrib/postgres/forms/jsonb/#JSONField
After (3.2.16): https://docs.djangoproject.com/en/3.2/_modules/django/forms/fields/#JSONField

The main change is to use the `decoder`.

```
try:
    json_value = json.loads(value, cls=self.decoder)
    return decode_datetime_objects(json_value)
except (json.JSONDecodeError, TypeError):
    raise ValidationError(
        self.error_messages['invalid'],
        code='invalid',
        params={'value': value},
    )
```

Reply:
ditto

```python
value = super(DateTimeAwareJSONFormField, self).to_python(value)
try:
    return decode_datetime_objects(value)
except TypeError:
    raise ValidationError(
        self.error_messages['invalid'],
        code='invalid',
        params={'value': value},
    )
```